src/cmd/windows/durable_task_monitor.go

```diff
@@ -0,0 +1,152 @@
+// +build windows
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"golang.org/x/sys/windows"
+        "jenkinsci.org/plugins/durabletask/common"
+)
+
+var logger *log.Logger
+
+type Shell int
+
+const (
+	CMD Shell = iota
+	POWERSHELL
+	PWSH
+)
+
+func (shell Shell) String() string {
+	switch shell {
+	case CMD:
+		return "cmd"
+	case POWERSHELL:
+		return "powershell"
+	case PWSH:
+		return "pwsh"
+	default:
+		return "UNKNOWN"
+	}
+}
+
+// Launches the script in a new session and waits for its completion.
+func launcher(wg *sync.WaitGroup, shell string, scriptPath string) {
+	defer wg.Done()
+
+	if _, err := os.Stat(scriptPath); err != nil {
+		if os.IsNotExist(err) {
+			logger.Printf("%s does not exist", scriptPath)
+			return
+		}
+	}
+
+	/*
+		outputFile, err := os.Create("output.txt")
+		if err != nil {
+			logger.Println(err.Error())
+			return
+		}
+		defer outputFile.Close()
+		errFile, err := os.Create("err.txt")
+		if err != nil {
+			logger.Println(err.Error())
+			return
+		}
+		defer errFile.Close()
+	*/
+
+	var cmd *exec.Cmd
+	switch shell {
+	case CMD.String():
+		cmd = exec.Command("cmd.exe", "/C", scriptPath)
```

Comment:
Is it guaranteed to be on the `C` drive?

Reply:
most likely not guaranteed

Reply:
dumb question: what type of failure are you concerned about? did you mean if `cmd` is guaranteed to be on the C drive? or the `scriptPath`?

Reply:
`/c` is the "run command" option,  not the drive.

Reply:
oh :facepalm: was not looking carefully, resolved